user_interface.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <string>
+#include "user_interface.h"
+#include <cmath>
+#include <iomanip> //std::setprecision
+
+using namespace std;
+//const ostream &standard = std::cout;
+const string white_line = "\011\012\013\014\015\040";
+
+void print_greeting(ostream &out)
+{
+    out << "Welcome to the random number generator!\n"
```

Comment:
Use `endl` with the streams. `\n` is for `printf`. Change it everywhere. To make it fast and automatic instead of changing every occurrence manually, let's learn using [`Regular Expressions`](https://en.wikipedia.org/wiki/Regular_expression) in `Find and Replace`. Try replacing "<<(.*?)\n\"" to "<<$1\"\ <<\ endl"
First, search using the first regular expression. You'll need to enable it in you IDE's search box (depends on your IDE, usually it's a small button by the search box, look for popup hints when you move your mouse cursor around). If you confirm that the search regex finds lines correctly, then try replacing one and see if the result is correct (I've written the regexes without trying them, so I might be wrong). Once you confirm it's correct, you can `replace all`.
```suggestion
    out << "Welcome to the random number generator!" << endl
```